R/total_collapse.R

```diff
@@ -0,0 +1,90 @@
+
+#' Collapse variables to single representation
+#'
+#' Simplify a data frame by collapsing specified variables, 
+#' according to the location of total codes, 
+#' into a single vector or by consolidating groups of variables into new columns.
+#'
+#' @param data A data frame containing the variables to be collapsed.
+#' @param variables A vector of variable names or a named list of variable names.
```

Comment:
Should it instead be "A vector of variable names or a list of named list of variable names."? This would better match the given example with named lists.

Reply:
Answer with help from ChatGPT: 

Thank you for the suggestion! I understand how using "vector" for "variable names" but not for "list" might seem inconsistent, especially since in R, both vectors and lists are technically vectors at a fundamental level.

However, in R's documentation and common usage, **"vector"** typically refers to atomic vectors (e.g., character, numeric, or logical vectors), while **"list"** is used specifically for recursive vectors, which can contain elements of different types, including other lists. This distinction is important to clarify the structure expected by the function.

For "variable names," the input is expected to be an atomic character vector, so **"vector"** is the appropriate term. In the second case, the input is a **list** of named groups of variables. For example:

```r
list(GEO = c("geo", "eu"), SECTOR = c("sector2", "sector4"))
```

Here, the list has two elements, each named ("GEO" and "SECTOR"), and each element is an atomic character vector. Referring to this as a "vector of lists" might imply additional structure or nesting that is not expected by the function.

By keeping this distinction, we align with R's conventions and make it clearer what the function expects.